code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -263,7 +263,14 @@
 						"<span class='notice'>You hug [src] to make [p_them()] feel better!</span>")
 			GET_COMPONENT_FROM(mood, /datum/component/mood, src)
 			if(mood)
-				mood.add_event("hug", /datum/mood_event/hug)
+				var/mob/living/L = src
+				if(L.has_trait(TRAIT_HOMOSEXUAL))
+					if(M.gender == L.gender)
+						mood.add_event("hug", /datum/mood_event/gayhug)
+					else
+						mood.add_event("hug", /datum/mood_event/badhug) //Gays are picky I guess?

```

Comment:
Kinda offensive my dude

Reply:
What if they are a neuter? Does that count?

Reply:
Everyone should enjoy hugs. Just give a stronger positive moodlet on a preferred-gender hug and a weaker moodlet on other hugs.

Reply:
@Iamgoofball you know damn well I'm the embodiment of everything soy and gay in the modern world and im not offended :^)